mock/src/main/java/com/tngtech/keycloakmock/impl/helper/TokenHelper.java

```diff
@@ -53,6 +57,25 @@ public String getToken(Session session, UrlConfiguration requestConfiguration) {
     return tokenGenerator.getToken(builder.build(), requestConfiguration);
   }
 
+  @Nullable
+  public String getServiceAccountToken(String clientId, UrlConfiguration requestConfiguration) {
```

Comment:
With the same reasoning as above, I'd argue that it is better in my opinion to keep the code simple and stupid and actually create an AdHocSession even for the client credentials grant, rather than build special logic for this case.